rust/arrow/src/record_batch.rs

```diff
@@ -216,15 +216,28 @@ impl Into<StructArray> for RecordBatch {
     }
 }
 
-/// Definition of record batch reader.
+/// Trait for types that can read `RecordBatch`'s in a single-threaded context.
 pub trait RecordBatchReader {
-    /// Returns schemas of this record batch reader.
-    /// Implementation of this trait should guarantee that all record batches returned
-    /// by this reader should have same schema as returned from this method.
+    /// Returns the schema of this `RecordBatchReader`.
+    ///
+    /// Implementation of this trait should guarantee that all `RecordBatch`'s returned by this
+    /// reader should have the same schema as returned from this method.
     fn schema(&mut self) -> SchemaRef;
 
-    /// Returns next record batch.
-    fn next_batch(&mut self) -> Result<Option<RecordBatch>>;
+    /// Reads the next `RecordBatch`.
+    fn next(&mut self) -> Result<Option<RecordBatch>>;
+}
+
+/// Trait for types that can read `RecordBatch`'s in a multi-threaded context.
+pub trait SendableRecordBatchReader: Send + Sync {
```

Comment:
Nitpick here because naming is hard and i don't have a good answer to this either ;P But the sendable prefix could be changed something more accurate since this trait also requires sync.

Reply:
I know I wasn't sure what to go with here, maybe `SendSyncRecordBatchReader`?  It's a bit verbose but probably the most accurate?

Reply:
I like `SendSync`, same length as Sendable, but more accurate. `Threaded` is another potential candidate as well. I wish there is a shorter word that conveys the same meaning as `Concurrent` :D

Reply:
I'll go with `SendSync` as it's the most explicit.  We can then see if there is further feedback from users.